packages/docprovider/src/tokens.ts

```diff
@@ -16,23 +17,11 @@ export const IDocumentProviderFactory = new Token<IDocumentProviderFactory>(
 /**
  * An interface for a document provider.
  */
-export interface IDocumentProvider {
+export interface IDocumentProvider extends IDisposable {
```

Comment:
Mind updating the extension migration guide with these API changes?

https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html#api-breaking-changes

Looks like the ones documented will not be up-to-date otherwise:

![image](https://user-images.githubusercontent.com/591645/198576808-20380484-f000-460a-88ba-1c33965740dc.png)

Reply:
ok then please open a new issue to track this for 4.0.